backend/admission.go

```diff
@@ -3,15 +3,16 @@ package backend
 import (
 	"context"
 
+	ctxHelpers "github.com/grafana/grafana-plugin-sdk-go/backend/context"
```

Comment:
Sorry if I missed this somewhere, but why the need for a new package for this?

Reply:
looks like it, yes, a cyclic dependency because of this [line](https://github.com/grafana/grafana-plugin-sdk-go/pull/1130/files#diff-f05f71d96061977c212825875fc9b3552e3f93cafa67fc869cac8c29b1b8fd51R137):

```
		labels["endpoint"] = ctxHelpers.EndpointFromContext(ctx).String()
```

My suggestion to avoid the breaking change:

 - Create a package under `backend/internal/endpointctx` and expose `EndpointCtxKeyType` and  `EndpointCtxKey`. Since it's within an `internal` folder, it won't be exposed outside of the repository.
 - Change `backend/endpoint.go` to use the above.
 - Change `backend/httpclient/datasource_metrics_middleware.go` and manually read the exposed key. Something like:

```go
		labels["endpoint"] = ""
		if ep := ctx.Value(endpointctx.EndpointCtxKey); ep != nil {
			labels["endpoint"] = ep.(string)
		}
```

Reply:
That sounds like a good alternative, I will implement and request a re-review 😊

Reply:
Nice @andresmgot - I think that'll do well for now. Ket's keep an eye on if this becomes more common, then we may look at a broader change to these packages perhaps.

Reply:
Implemented changes 😄